### DISCUSSION TITLE
How can I use create_new_note_from_template in dataviewjs?

### DISCUSSION THREAD

**Discussion by Another5405** [SilentVoid13/Templater#973] [Help] [answered]
Hello,
I was try to create note using templater and dataviewjs
but my result spawn empty note only,(the template "FakeTem" have some text)
here is my code:
```
const path = dv.page("FakeTem").file.path;
const file = this.app.vault.getAbstractFileByPath(path);

const buttonMaker = (file) => {
    const btn = this.container.createEl('button', {"text": "Test"});
    const {create_new_note_from_template} = this.app.plugins.plugins["templater-obsidian"].templater;
    btn.addEventListener('click', async (evt) => {
        evt.preventDefault();
        await create_new_note_from_template(file);
    });
    return btn;
}

buttonMaker();

//FakeTem.md
dv.el("b",path)
//<t>
dv.el("b",file)

```

I also try write_template_to_file or add more property in function but the new note is empty too.

ThankYou so much.

**Comment by AB1908** [comment]
I'm not sure you're using the templater functions correctly. Regardless, what's the use case?

**Reply by Another5405** [reply]
Generate record by one click,
Almost use for bridge two type of note.

**Reply by Another5405** [reply]
Thank you for always reminding me to use the console. 
I thought console which is the dataview bug report and I face error over again and again without the real console,
And finally I know were to open it, Ctrl Shift I is alway not working and lucky someone mention this problem these day.

**Comment by Another5405** [comment, marked as answer]
A little tricky bug if you don't know how to open console...
I open the console tool first time.

this is the code which is work.
```dataviewjs
const path = dv.page("FakeTemplater").file.path;
const file = this.app.vault.getAbstractFileByPath(path);
const templater = this.app.plugins.plugins["templater-obsidian"].templater;
const buttonMaker = () => {
    const btn = this.container.createEl('button', {"text": "Test"});

    btn.addEventListener('click', async (evt) => {
        evt.preventDefault();
        await templater.create_new_note_from_template(file,null,"A",true);
    });
    return btn;
}

buttonMaker();

//FakeTem.md
dv.el("b",path)
//<t>
dv.el("b",file)

```
